src/lib/wicked/connections.js

```diff
@@ -72,20 +72,22 @@ const addressesFromConfig = (type, addresses) => {
     return addresses.map(addr => {
         const { local, label } = addr;
         return model.createAddressConfig(
-            { type, label, address: local }
+            { type, local, label }
         );
     });
 };
 
-const ipConfig = (config) => {
+const ipConfig = (config, type = 'ipv4') => {
     let addresses = [];
     let bootProto;
+    const wickedAddresses = config[`${type}:static`];
+    const wickedDhcp = config[`${type}:dhcp`];
 
-    if (config['ipv4:static']?.addresses) {
-        addresses = addressesFromConfig(addressType.IPV4, config['ipv4:static'].addresses);
+    if (wickedAddresses) {
+        addresses = addressesFromConfig(addressType.IPV4, wickedAddresses);
```

Comment:
If I got it correctly, we should use the `type`here too. I mean, 

```js
addresses = addressesFromConfig(addressType[type.toUpperCase()], wickedAddresses);
```

or similar.